### DISCUSSION TITLE
kind version didn't match the latest version

### DISCUSSION THREAD

**Discussion by charles-chenzz** [Homebrew/discussions#4650] [Getting started]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20
ORIGIN: https://mirrors.ustc.edu.cn/brew.git/
HEAD: 344d32bf7f40183359592ef81ca6486e0587981b
Last commit: 6 months ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: d258972e74660e248af87ad4f00b654645f6648e
Core tap last commit: 10 months ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.37.3 => /opt/homebrew/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.3-arm64
CLT: 13.4.0.0.1.1651278267
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Warning: Suspicious https://github.com/Homebrew/brew git origin remote found.
The current git origin is:
  https://mirrors.ustc.edu.cn/brew.git/

With a non-standard origin, Homebrew won't update properly.
You can solve this by setting the origin remote:
  git -C "/opt/homebrew" remote set-url origin https://github.com/Homebrew/brew

Warning: Suspicious https://github.com/Homebrew/homebrew-core git origin remote found.
The current git origin is:
  https://mirrors.ustc.edu.cn/homebrew-core.git/

With a non-standard origin, Homebrew won't update properly.
You can solve this by setting the origin remote:
  git -C "/opt/homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin https://github.com/Homebrew/homebrew-core

Warning: /usr/bin occurs before /opt/homebrew/bin in your PATH.
This means that system-provided programs will be used instead of those
provided by Homebrew. Consider setting your PATH so that
/opt/homebrew/bin occurs before /usr/bin. Here is a one-liner:
  echo 'export PATH="/opt/homebrew/bin:$PATH"' >> ~/.zshrc

The following tools exist at both paths:
  git-receive-pack
  bashbug
  git-upload-archive
  git-shell
  git-upload-pack
  git
```

### Description of issue

I have app ``kind`` install by brew, and I want to upgrade it to latest version, which is already merge https://github.com/Homebrew/homebrew-core/pull/133771

but I run ``brew upgrade kind``, it still keeps in 0.5, then I uninstall and install again, it's still in 0.5. just want to know if I've missing something?

**Comment by gromgit** [comment]
Your `brew config` and `brew doctor` output show some unusual circumstances:
1. You're behind the Great Firewall of China, so you found the need to use a **China-based Homebrew mirror**.
2. You have a **6-month-old Homebrew v3.x software setup**, so you're forced to use on-disk Homebrew taps instead of the v4.x JSON API capability that give most other Homebrew users the latest and greatest updates without on-disk taps.
3. You have a **10-month-old Homebrew core tap**, so you'll never get the latest and greatest software.
4. `Warning: /usr/bin occurs before /opt/homebrew/bin in your PATH.` That usually makes no sense in most user environments, and if you have a `kind` binary in `/usr/bin`, you've just forced yourself to run the _system-level_ `kind`, no matter what Homebrew version you install.

Things to check:
1. `type -p kind` -- are you running the wrong `kind` binary?
2. `brew info kind` -- which version is actually available from your core tap?
3. Run `brew update`, then `brew config` to see if you now have more recent "last commit" values. If you're still stuck with "_N_ months ago", look carefully at the `brew update` output for errors. You **MUST** fix this issue, otherwise you're stuck in "Homebrew past" forever.

**Comment by charles-chenzz** [comment]
<img width="962" alt="image" src="https://github.com/Homebrew/discussions/assets/73322208/89231751-6662-4289-b006-463db37de715">

the brew update works, and seems like now I can install latest version of kind
